### ISSUE TITLE
Reduce calls to should_disable

### ISSUE THREAD

**Issue by jvoisin** [nbs-system/snuffleupagus#166]
Currently, this function (along with its callee `get_complete_funtion_path`) is our main bottleneck. We shouldn't call it twice for "native" functions.

Ideally, we should split the list of functions to hook into two lists: one for functions to hook at runtime, and the other for functions that we manager to hook at loading time. The later could even be a hashtable or lists, but maybe it's a bit overkill, I don't know.

Or we could, for each "native" function, put the names of the ones we're hooking into a hashtable, so when we're entering the hook of one, we can simply check into the hashtable if it's here, and walk the linked-list  if it is. If we're doing this, we could split the functions into several lists.

**Comment by xXx-caillou-xXx** [comment]
I did some benchmarks using joomla's phpunit on the `should_be_faster` branch. It seems to be a *bit* faster.
I used 24 rules from the `default.rules` file and added X line of rules with random names.

number of rules | time on `should_be_faster` | time on `master`
--- | --- | ---
25 | 0m51.645s | 0m50.151s
50 | 0m51.748s | 0m52.506s
100 | 0m51.750s | 0m57.207
500 | 0m51.963s | 1m41.236s
1000 | 0m53.306 | 2m55.564s

without snuffleupagus: 0m41.298s
with snuffleupagus and every functions hooked at startup: 0m41.736s
